### PR TITLE
Final CICD Port

### PR DESCRIPTION
**Pull request by sweeneyskirt-sl**
### Description
Final porting of Cookbook CICd topics -- Bitbucket, Bamboo, TeamCity, Azure.

### Motivation and Context
Migrating content.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [ ] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/master/CONTRIBUTING.MD) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/ci/teamcity.md [hunk 1]
```diff
@@ -5,4 +5,93 @@ sidebar_label: TeamCity
...
 ---
 
-Sauce Labs TeamCity page.
+[TeamCity](https://www.jetbrains.com/teamcity/) is a continuous integration tool suite by JetBrains. Use this documentation to manage your Sauce Labs tests within your TeamCity Java project.
+
+## Installing the Plugin
+
+1. Download the [Sauce OnDemand Plugin for TeamCity](https://saucelabs.com/downloads/teamcity/release/com/saucelabs/teamcity/build/1.48/build-1.48.zip) ZIP file.
+1. Copy the ZIP into your TeamCity `~/.BuildServer/plugins` directory and extract the files.
+1. Restart TeamCity.
```
Comment (spider-sauce) [inline docs/ci/teamcity.md:14]:
```suggestion
1. Download the [Sauce OnDemand Plugin for TeamCity](https://saucelabs.com/downloads/teamcity/release/com/saucelabs/teamcity/build/1.48/build-1.48.zip) ZIP file.
2. Copy the ZIP into your TeamCity `~/.BuildServer/plugins` directory and extract the files.
3. Restart TeamCity.
```

File: docs/ci/teamcity.md [hunk 2]
```diff
@@ -5,4 +5,93 @@ sidebar_label: TeamCity
...
+
+## Configuring a Project
+
+### Create the Project
+
+1. From the TeamCity dashboard, choose  **Administration** and click **Create Project**.
+1. For **Name**, enter `SauceDemo` (populates **Project ID** field with `SauceDemo`.
+1. Click **Create**.
+1. Select the **VCS Roots** tab, click **Create VCS root**, and enter the following field values:
```
Comment (spider-sauce) [inline docs/ci/teamcity.md:23]:
```suggestion
1. From the TeamCity dashboard, choose  **Administration** and click **Create Project**.
2. For **Name**, enter `SauceDemo` (populates **Project ID** field with `SauceDemo`.
3. Click **Create**.
4. Select the **VCS Roots** tab, click **Create VCS root**, and enter the following field values:
```

File: docs/ci/teamcity.md [hunk 3]
```diff
@@ -5,4 +5,93 @@ sidebar_label: TeamCity
...
+1. In **Attach existing VCS root**, select `https://github.com/saucelabs/java-sauce-example.git#master`.
+1. Click **Add build step**.
+    * For **Runner type**, choose `Maven`.
+    * For **Goals**, enter `test`.
+    * **Save** the build step.
+1. Click **Add build feature** and select **Sauce Labs Build Feature**.
+1. Enter your `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY`.
+1. Select **Enable Sauce Connect** if you want to launch an instance of Sauce Connect prior to running your job that will close upon completion of the job (for use in private network environments in order to establish a secure connection to the Sauce platform).
+1. Select the operating system and browser combination you want to test against.  
+1. Click **Save**.
```
Comment (spider-sauce) [inline docs/ci/teamcity.md:42]:
```suggestion
1. On the **General** tab, choose **Create build configuration**.
2. In the **Name** field, enter `Maven`.
3. Click **VCS settings**.
4. In **Attach existing VCS root**, select `https://github.com/saucelabs/java-sauce-example.git#master`.
5. Click **Add build step**.
    * For **Runner type**, choose `Maven`.
    * For **Goals**, enter `test`.
    * **Save** the build step.
6. Click **Add build feature** and select **Sauce Labs Build Feature**.
7. Enter your `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY`.
8. Select **Enable Sauce Connect** if you want to launch an instance of Sauce Connect prior to running your job that will close upon completion of the job (for use in private network environments in order to establish a secure connection to the Sauce platform).
9. Select the operating system and browser combination you want to test against.  
10. Click **Save**.
```

File: docs/ci/teamcity.md [hunk 4]
```diff
@@ -5,4 +5,93 @@ sidebar_label: TeamCity
...
+
+### Integrate the Tests
+
+1. From the TeamCity dashboard, click **Run**.
+1. When the build completes, click **Results** and choose the **Sauce Labs Results** tab.
+1. Click the **Job ID** link to view the test report, which lists the steps performed the test and includes a video of the test.
```
Comment (spider-sauce) [inline docs/ci/teamcity.md:48]:
```suggestion
1. From the TeamCity dashboard, click **Run**.
2. When the build completes, click **Results** and choose the **Sauce Labs Results** tab.
3. Click the **Job ID** link to view the test report, which lists the steps performed the test and includes a video of the test.
```

**Review by spider-sauce — COMMENTED** [review]
@sweeneyskirt-sl not sure why some of the numbered lists have `1.` listed for all entries. Is that to avoid markdown formatting issues once it is rendered? Not a big deal but I called out some  commit suggestions.

**Comment by sweeneyskirt-sl** [comment]
Hi @spider-sauce -- Thanks for reviewing! So, I guess I am old school. I have always learned that you are supposed to only use all `1.`s in Markdown. That way you never have to keep track of your numbering and it will always be correct when it is rendered. I thought that was one of the "rules" of Markdown.

**Comment by spider-sauce** [comment]
> Hi @spider-sauce -- Thanks for reviewing! So, I guess I am old school. I have always learned that you are supposed to only use all `1.`s in Markdown. That way you never have to keep track of your numbering and it will always be correct when it is rendered. I thought that was one of the "rules" of Markdown.

Huh, I didn't know that was a thing. Either way, not a show stopper so you can leave it if you want and just merge 👍

**Review by spider-sauce — APPROVED** [review]

**Pull request merged**